aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -24,6 +24,48 @@
 #include <numeric>
 
 namespace at {
+
+namespace meta {
+
+TORCH_META_FUNC(addmm)(const Tensor &self, const Tensor &mat1, const Tensor &mat2,
+                       const Scalar& beta, const Scalar& alpha) {
+  TORCH_CHECK(mat1.dim() == 2, "mat1 must be a matrix, got ",
+              mat1.dim(), "-D tensor");
+  TORCH_CHECK(mat2.dim() == 2, "mat2 must be a matrix, got ",
+              mat2.dim(), "-D tensor");
+
+  const auto self_sizes = self.sizes();
+  auto m1_sizes = mat1.sizes();
+  auto m2_sizes = mat2.sizes();
+
+  if (self.numel() != 0) {
+    // if number of elements are zero, only then can the input tensor can be resized.
+    TORCH_CHECK((self.dim() == 2), "input matrix must be a matrix, got ",
+                self.dim(), "-D tensor");
+    TORCH_CHECK(((self_sizes[0] == m1_sizes[0]) && (self_sizes[1] == m2_sizes[1])),
+                "The input matrix must have the same size as the product of mat1 & mat2");
+  }
```

Comment:
**Unstructured implementation:**
CUDA - Resizes input tensor, except for the inplace variant
CPU - Resizes the input tensor in all cases

**TODO:** 
Remove this check. This check should only be run for the inplace variant. However, meta combines everything, so this check is being run for all variants. Figure out a way to test it only for the inplace variant.